pkgs/intl4x/lib/src/datetime_format/datetime_format_4x.dart

```diff
@@ -66,6 +47,43 @@ class DateTimeFormat4X extends DateTimeFormatImpl {
               )
             : null;
 
+  static icu.DateTimeFormatter? _setDateTimeFormatter(
+    DateTimeFormatOptions options,
+    Data data,
+    Locale locale,
+  ) {
+    final dateFormatStyle = options.dateFormatStyle;
+    final timeFormatStyle = options.timeFormatStyle;
+    if (dateFormatStyle != null && timeFormatStyle != null) {
+      return icu.DateTimeFormatter.withLengths(
+        data.to4X(),
+        locale.to4X(),
+        dateFormatStyle.dateTo4xOptions(),
+        timeFormatStyle.timeTo4xOptions(),
+      );
+    } else {
+      return null;
+    }
```

Comment:
go/tott/651, does Dart allow you to do:
```suggestion
    if (dateFormatStyle != null && timeFormatStyle != null) {
      return null;
    }
    
    return icu.DateTimeFormatter.withLengths(
      data.to4X(),
      locale.to4X(),
      dateFormatStyle.dateTo4xOptions(),
      timeFormatStyle.timeTo4xOptions(),
    );
```